### PR TITLE
Update docs for the new dataset API.

### PR DESCRIPTION
**Pull request by ChrisCummins**
This is the sibling of #222. It updates the documentation of the getting started guide, tutorials, API reference etc to the new dataset API.

In general, this means simplifying things, as we no longer need to explain how to download and manage datasets.

Issue #45.

### CODE REVIEW

File: compiler_gym/bin/service.py [hunk 1]
```diff
@@ -20,47 +20,52 @@
...
+    | AutophaseDict            | `Dict(ArgsPhi:int<0,inf>, BB03Phi:int<0,...` |
+    +--------------------------+----------------------------------------------+
+    | BitcodeFile              | `str_list<>[0,4096.0])`                      |
+    +--------------------------+----------------------------------------------+
     ...
 
-The capabilities of an unmanaged service can be queried using the
-:code:`--service` flag. For example, query a service running at
-:code:`localhost:8080`:
+The output is a reStructuredText document containing tabular summaries of the
```
Comment (hughleat) [inline compiler_gym/bin/service.py]: What is a `reStructuredText` document? Can you provide a link?

File: docs/source/getting_started.rst [hunk 2]
```diff
@@ -176,31 +156,31 @@ The upper and lower bounds of the reward signal are described by
...
-:attr:`env.benchmark <compiler_gym.envs.CompilerEnv.benchmark>`:
+    array([  0,   4,  54,  39,  12,  46,  23,   6,  12,  31,   2,   4,   0,
+            81,   4,  77,  13,  15, 108, 106,  75,  51,  71,  46,  15,   0,
+             9,  46,   0,  13,  72,  51,  77,  81,  39,  31,   0, 163,   2,
+             0,   4,   6,  13,   1,   0,  73,   8,   1,   0,  15,  85, 638,
+           402,  16,  10, 298])
+
+The numpy array that is returned by :meth:`reset()
+<compiler_gym.envs.CompilerEnv.reset>` is the initial observation of the program
+state. This value, along with the entire dynamics of the environment, depend on
```
Comment (hughleat) [inline docs/source/getting_started.rst]: `depend` -> `depends`

File: docs/source/getting_started.rst [hunk 3]
```diff
@@ -331,65 +319,98 @@ And to describe the capabilities of each environment:
...
+    ------------------
+
+    +--------------------------+----------------------------------------------+
+    | Observation space        | Shape                                        |
+    +==========================+==============================================+
+    | Autophase                | `Box(0, 9223372036854775807, (56,), int64)`  |
+    +--------------------------+----------------------------------------------+
+    | AutophaseDict            | `Dict(ArgsPhi:int<0,inf>, BB03Phi:int<0,...` |
+    +--------------------------+----------------------------------------------+
+    | BitcodeFile              | `str_list<>[0,4096.0])`                      |
```
Comment (hughleat) [inline docs/source/getting_started.rst:345]: Are these tables duplicated in multiple files?
Is there someway to combine them so if someone makes a change they will only have to update in one place?

**Review by hughleat — APPROVED** [review]
LGTM

File: compiler_gym/bin/service.py (hunk 1 above)
Comment (ChrisCummins) [reply compiler_gym/bin/service.py]: Done

File: docs/source/getting_started.rst (hunk 2 above)
Comment (ChrisCummins) [reply docs/source/getting_started.rst]: Good catch, done

File: docs/source/getting_started.rst (hunk 3 above)
Comment (ChrisCummins) [reply docs/source/getting_started.rst:345]: Sort of. There are a couple of illustrative examples in docstrings, but the values don't matter. The actual LLVM datasets table is in `llvm/index.rst` and is generated by the `compiler_gym.bin.service` script and copy-pasted into the docs. It does require a bit of effort to keep them in sync, but I think automating that would be OTT

**Pull request merged**